api/v1alpha1/componentsubscription_types.go

```diff
@@ -61,16 +61,22 @@ type OCMRepository struct {
 
 // ComponentSubscriptionStatus defines the observed state of ComponentSubscription
 type ComponentSubscriptionStatus struct {
-	// LatestVersion defines the version that was last reconciled successfully.
+	// LatestVersion defines the latest version encountered while checking component versions.
+	// This might be different from Replicated version which should be the latest applied/replicated version.
+	// The difference might be caused because of semver constraint or failures during replication.
 	LatestVersion string `json:"latestVersion"`
 
 	// ObservedGeneration is the last reconciled generation.
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 
+	// ReplicatedVersion defines the final version that has been applied to the destination component version.
 	//+optional
 	ReplicatedVersion string `json:"replicatedVersion,omitempty"`
```

Comment:
Hmm, the difference between ReplicatedVersion and LatestVersion is not clear to me.

Maybe it should just be similar to flux: 
- `lastAttemptedVersion
- lastAppliedVersion

?

Reply:
Yep I like that. Good idea.